packages/eslint-plugin/src/util/isTypeImport.ts

```diff
@@ -0,0 +1,29 @@
+import type {
+  Definition,
+  ImportBindingDefinition,
+} from '@typescript-eslint/scope-manager';
+import { DefinitionType } from '@typescript-eslint/scope-manager';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+/**
+ * Determine whether a variable definition is a type import. e.g.:
+ *
+ * ```ts
+ * import type { Foo } from 'foo';
+ * import { type Bar } from 'bar';
+ * ```
+ *
+ * @param definition - The variable definition to check.
+ *
+ * @returns boolean
```

Comment:
[Docs] This is an example of when explicit type annotations / comments are not very helpful. This function is a type predicate; explicitly saying `@returns boolean` is _less_ information than that.

```suggestion
```